BasicSample/Assets/AppRemoting/Scripts/AppRemotingSample.cs

```diff
@@ -0,0 +1,298 @@
+
+            ShowConnection2DUI();
+        }
+
+        private void Update()
+        {
+            var ip = textInput.text;
+            var hostIp = GetLocalIPAddress();
+            var connectPort = remotingConfiguration.RemotePort;
+            var listenPort = remotingListenConfiguration.TransportListenPort;
+            var connectionStateValid = Remoting.AppRemoting.TryGetConnectionState(out Remoting.ConnectionState connectionState, out Remoting.DisconnectReason disconnectReason);
+
+            if (connectionStateValid)
+            {
+                if (m_connectionState != connectionState || disconnectReason != m_disconnectReason)
+                {
+                    m_connectionState = connectionState;
+                    m_disconnectReason = disconnectReason;
+
+                    if(m_appRemotingMode == AppRemotingMode.connect)
+                    {
+                        Debug.Log($"Connection state changed : {ip}:{connectPort}, {connectionState}, {m_disconnectReason}");
+                    }
+                    else if (m_appRemotingMode == AppRemotingMode.listen)
+                    {
+                        Debug.Log($"Connection state changed : {hostIp}:{listenPort}, {connectionState}, {m_disconnectReason}");
+                    }
+                    
+
+                    switch (m_connectionState)
+                    {
+                        case Remoting.ConnectionState.Connected:
+                            HideConnection2DUI();
+                            break;
+                        case Remoting.ConnectionState.Connecting:
+                            ShowConnection2DUI();
+                            break;
+                        case Remoting.ConnectionState.Disconnected:
+                            ShowConnection2DUI();
+                            break;
+                    }
+                }
+            }
+            else
+            {
+                m_connectionState = Remoting.ConnectionState.Disconnected;
+            }
+
+            string commonMessage = "Welcome to App Remoting! Provide Ip address & click Connect or click Listen";
+
+            string connectMessage = string.IsNullOrWhiteSpace(ip)
+                    ? $"No IP address was provided to {nameof(Remoting.AppRemoting)}."
+                        : m_connectionState == Remoting.ConnectionState.Connected
+                            ? $"Connected to {ip}:{connectPort}."
+                            : m_connectionState == Remoting.ConnectionState.Connecting
+                                ? $"Connecting to {ip}:{connectPort}..."
+                                : $"Disconnected to {ip}:{connectPort}. Reason is {m_disconnectReason}";
+            string listenMessage = m_connectionState == Remoting.ConnectionState.Connected
+                            ? $"Connected on {hostIp}."
+                            : m_connectionState == Remoting.ConnectionState.Disconnected && m_disconnectedOnListenMode
```

Comment:
suggest to replace with a state "m_listening" to wrap around the "startcorouting" in listen function

Reply:
This issue did not resolve with m_listening since there is no way to find the end of start Listening coroutine for now, once that is in place this will be modified. Here is a new task for the API modification - #903